src/components/walletconnect/HeaderWidget/index.tsx

```diff
@@ -9,7 +8,10 @@ import { useWalletConnectSearchParamUri } from '@/services/walletconnect/useWall
 import Icon from './Icon'
 import SessionManager from '../SessionManager'
 import Popup from '../Popup'
-import { SuccessBanner } from '../SuccessBanner'
+import { ConnectionBanner } from '../ConnectionBanner'
+import useSafeInfo from '@/hooks/useSafeInfo'
+import { isUnsupportedChain } from '@/services/walletconnect/utils'
+import { useDeferredListener } from '@/hooks/useDefferedListener'
```

Comment:
GitHub might be hiding some changes but I don't see useDeferredListener used anywhere.

Reply:
It's used in `useSuccessSession` (L35) and `useDeleteSession` (L45).

Reply:
I see, thanks. IMHO this hook is wrapping some trivial logic in an untrivial way with cryptic paramas, so I would not add it. Let's merge it but I will refactor this later.